### PR TITLE
Added a square root button and unit tests to test its functionality.

### PR DESCRIPTION
**Pull request by edockery**
iosunittestssquareroot
screen shot 2016-09-12 at 11 09 39 am

### CODE REVIEW

**Comment by nielubowicz** [comment]
To begin, I think this unit test is invalid because [The Square root of 0 is generally accepted to be 0](https://en.wikipedia.org/wiki/Square_root#Principal_square_roots_of_the_positive_integers), not `NaN` or ERROR. As well, I think there is an error in the [code for the numeric value](https://github.com/Mobiquity/iOSUnitTests/pull/12/files#diff-186100947cade63b7898b0b420cc3d67R87) as positive real numbers, not just integers, have square roots.

As well, see this GIF about ERROR text vs. the calculator handling NaN: I think the best solution would be to make a new operation and use the chain of operands to get better error handling.
As well, see this GIF about ERROR text vs. the calculator handling NaN

Finally, (and please come find me if you get stuck on this), the code which runs the operation is not very well testable, as:
- you are injecting a `nil` sender (which is a code smell)
- I am able to break your unit test by changing one character nowhere the lines of code you added -- can you guess what it was and why?
- a core tenet of unit testing is that operations, functions, data storage, etc should all be _small, testable portions of code_ -- neatly packable for us as objects -- that can be manipulated to test edge cases and test against a variety of behaviors. Essentially, you want to be able to put your test probes _right up to_ your object and test it, without other objects (i.e. an `MCACalculatorViewController`) getting in your way.

For a square root function, I would like to see at least the following unit tests, verifying a small piece of output (not the contents of a label).
- `NaN` or invalid input
- negative number input
- positive integer input
- positive real input
- zero input

**Comment by edockery** [comment]
This is amazing feedback and very helpful. I will implement these changes and try again. 
Thank you.

**Comment by edockery** [comment]
Added four tests for the new operation function that was implemented as well as changing the code for the IBAction to just call the MCACalculator function getSquareRoot.

File: Calculator/CalculatorTests/Controllers/MCACalculatorViewControllerTests.m [hunk 1]
```diff
@@ -25,7 +25,7 @@ - (IBAction)numericInputButtonTapped:(id)sender;
 - (IBAction)clearButtonTapped:(id)sender;
 - (IBAction)operatorButtonTapped:(UIButton *)sender;
 - (NSMutableArray *)createOperators;
-
+- (IBAction)squarerootButtonTapped:(id)sender;
```
Comment (nielubowicz) [inline Calculator/CalculatorTests/Controllers/MCACalculatorViewControllerTests.m]: Does this still need to be here?

**Comment by edockery** [comment]
Removed unused code from test case.

File: Calculator/Calculator/MCACalculator.h [hunk 2]
```diff
@@ -20,5 +20,6 @@
 - (NSDecimalNumber *)evaluateExpressionFromHistory;
 
 - (void)clearAllCalculatorHistory;
-
+- (NSString *)getSquareRoot: (NSString *)calculatorDisplayLabel;
+-(void)squareRootError;
```
Comment (nielubowicz) [inline Calculator/Calculator/MCACalculator.h]: Please remove unused header method. As well, I'd prefer a different title for `getSquareRoot`'s parameter -- it's no longer the `calculatorDisplayLabel`, is it ;)?

**Comment by nielubowicz** [comment]
Approved. Great job, @edockery!
